### DISCUSSION TITLE
Netgear ifOperStatus OID - No Data

### DISCUSSION THREAD

**Discussion by fontanafluke** [prometheus/snmp_exporter#1263] [Q&A]
### Host operating system: output of `uname -a`
Linux  5.14.0-480.el9.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Jul 12 20:45:27 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
### snmp_exporter version: output of `snmp_exporter -version`
snmp_exporter, version 0.26.0 (branch: HEAD, revision: 44f8732988e726bad3f13d5779f1da7705178642)
  build user:       root@90ba0aabb239
  build date:       20240511-11:16:35
  go version:       go1.22.3
  platform:         linux/amd64
  tags:             unknown

### What device/snmpwalk OID are you using?
Netgear     - 1.3.6.1.2.1.2.2
### What did you do that produced an error?
No error, debug says it walks successfully, but I never see data propagate in Prometheus.

**Comment by fontanafluke** [comment]
Hello,

Any help would be greatly appreciated.

Here are some more details. I have scoured the forums and am not finding the right breadcrumb to lead me to the solution.

-I downloaded the official MIB file from Netgear directly. 
-I generated the yml using the generator tool and appended it to the default snmp.yml that comes with the install. Output below.

./generator generate -m /root/mibs/ -o /root/snmp.yml -g generator.yml
ts=2024-10-10T13:06:45.625Z caller=net_snmp.go:175 level=info msg="Loading MIBs" from=/root/mibs/
ts=2024-10-10T13:06:45.715Z caller=main.go:58 level=info msg="Generating config for module" module=netgear
ts=2024-10-10T13:06:45.721Z caller=main.go:139 level=error msg="Error generating config netsnmp" err="cannot find oid '1.3.6.1.2.1.1.1.0' to walk"

Note: For some reason it does not include the OIDs in the walk section when generated. I have tried manually adding them for testing.

-It pulls data from most OIDs successfully.
-I ran the debug and it says it walks the OID successfully.
-A manual snmpwalk successfully pulls information as expected.

I have attached the generator.yml and modified snmp.yml output that I am using.

[generator.yml.txt](https://github.com/user-attachments/files/17328660/generator.yml.txt)
[snmp.yml.txt](https://github.com/user-attachments/files/17329370/snmp.yml.txt)


Thanks!

**Reply by bastischubert** [reply]
snmp.yml looks reasonable - whats the output you get when you manually scrape the snmp endpoint of the snmp_exporter and what does the log say ( try ./snmp_exporter --log.level=debug --snmp.debug-packets )
